src/HexViewer.ts

```diff
@@ -177,12 +181,27 @@ export class HexViewer {
                 e.key === "ArrowRight" ? 1 : e.key === "ArrowLeft" ? -1 : null;
 
             if (selDiff === null) return;
-
-            var newSel = this.selectionStart + selDiff;
+            var newSel = this.selectionDragEnd + selDiff;
             if (newSel < 0) newSel = 0;
             else if (newSel >= this.dataProvider.length) newSel = this.dataProvider.length - 1;
-
-            this.setSelection(newSel);
+            if (!e.shiftKey) {
+                this.selectionDragStart = newSel;
+                this.setSelection(newSel);
+            } else {
+                var smaller = Math.min(
+                    this.selectionDragStart,
+                    newSel
+                );
+                var larger = Math.max(
+                    this.selectionDragStart,
+                    newSel
+                );
+                this.setSelection(
+                    smaller,
+                    larger
+                );
```

Comment:
This is unnecessary as well, because the method `setSelection` already does it:

https://github.com/kaitai-io/kaitai_struct_webide/blob/79a30577e8dac208b4dfac92575a9ce3f9f42a96/src/HexViewer.ts#L321-L322

So this will be enough:

```suggestion
                this.setSelection(
                    this.selectionDragStart,
                    newSel
                );
```